cli/commands/project_sync.go

```diff
@@ -0,0 +1,249 @@
+
+type Volume struct {
+	External bool
+}
+
+const UNISON_PORT = 5000
+const MAX_WATCHES = "100000"
+
+func (cmd *ProjectSync) Commands() []cli.Command {
+	start := cli.Command{
+		Name:        "sync:start",
+		Aliases:     []string{"sync"},
+		Usage:       "Start a unison sync on local project directory. Optionally provide a volume name.",
+		ArgsUsage:   "[optional volume name]",
+		Description: "Volume name will be discovered in the following order: argument to this command > outrigger project config > docker-compose file > current directory name",
+		Before:      cmd.Before,
+		Action:      cmd.RunStart,
+	}
+	stop := cli.Command{
+		Name:        "sync:stop",
+		Usage:       "Stops a unison sync on local project directory. Optionally provide a volume name.",
+		ArgsUsage:   "[optional volume name]",
+		Description: "Volume name will be discovered in the following order: argument to this command > outrigger project config > docker-compose file > current directory name",
+		Before:      cmd.Before,
+		Action:      cmd.RunStop,
+	}
+
+	return []cli.Command{start, stop}
+}
+
+// Start the unison sync process
+func (cmd *ProjectSync) RunStart(ctx *cli.Context) error {
+	project.ConfigInit()
+	config := project.GetProjectConfigFromFile(project.GetConfigPath())
+	volumeName := cmd.GetVolumeName(ctx, config)
+	cmd.out.Verbose.Printf("Starting sync with volume: %s", volumeName)
+
+	// Ensure the processes can handle a large number of watches
+	if err := cmd.machine.SetSysctl("fs.inotify.max_user_watches", MAX_WATCHES); err != nil {
+		cmd.out.Error.Fatalf("Error configuring file watches on Docker Machine: %v", err)
+	}
+
+	cmd.out.Info.Printf("Starting sync volume: %s", volumeName)
+	exec.Command("docker", "volume", "create", volumeName).Run()
+
+	cmd.out.Info.Println("Starting unison container")
+	unisonMinorVersion := cmd.GetUnisonMinorVersion()
+	cmd.out.Verbose.Printf("Local unison version for compatibilty: %s", unisonMinorVersion)
+	exec.Command("docker", "container", "stop", volumeName).Run()
+	err := exec.Command("docker", "container", "run", "--detach", "--rm",
+		"-v", fmt.Sprintf("%s:/unison", volumeName),
+		"-e", "UNISON_DIR=/unison",
+		"-l", fmt.Sprintf("com.dnsdock.name=%s", volumeName),
+		"-l", "com.dnsdock.image=volume.outrigger",
+		"--name", volumeName,
+		fmt.Sprintf("outrigger/unison:%s", unisonMinorVersion),
+	).Run()
+	if err != nil {
+		cmd.out.Error.Fatalf("Error starting sync container %s: %v", volumeName, err)
+	}
+
+	var ip = cmd.WaitForUnisonContainer(volumeName)
+
+	cmd.out.Info.Println("Initializing sync")
+
+	// Remove the log file, the existence of the log file will mean that sync is up and running
+	var logFile = fmt.Sprintf("%s.log", volumeName)
+	exec.Command("rm", "-f", logFile).Run()
+
+	unisonArgs := []string{
+		".",
+		fmt.Sprintf("socket://%s:%d/", ip, UNISON_PORT),
+		"-auto", "-batch", "-silent", "-contactquietly",
+		"-repeat", "watch",
+		"-prefer", ".",
+		"-logfile", logFile,
+		"-ignore", "Name .git",
```

Comment:
Shouldn't there be some processing to combine this with the sync configuration for ignores I see in the project.ProjectSync struct?